src/content/blog/astro-5100.mdx

```diff
@@ -262,6 +263,64 @@ export default defineConfig({
 
 For more details, see the [experimental CSP documentation](https://docs.astro.build/en/reference/experimental-flags/csp/).
 
+## Customizable Cloudflare Workers entrypoint
+
+Some Cloudflare Workers features, like [Durable Objects](https://developers.cloudflare.com/workers/runtime-apis/durable-objects/), [Cloudflare Queues](https://developers.cloudflare.com/queues/), and [Cron Triggers](https://developers.cloudflare.com/workers/platform/cron-triggers/) require a custom entrypoint file. Because the `@astrojs/cloudflare` adapter generates the entrypoint used for serving a site, these features have previously not been possible to use with Astro. Astro 5.10 adds support for customizing the entrypoint file used by the `@astrojs/cloudflare` adapter, giving you full control over your Cloudflare Workers setup.
+
+To customize the entrypoint, add the `workerEntryPoint` option to your `@astrojs/cloudflare` adapter configuration in your `astro.config.mjs` file:
+
+```js title="astro.config.mjs" {6-9}
+import { defineConfig } from 'astro/config';
+import cloudflare from '@astrojs/cloudflare';
+
+export default defineConfig({
+	adapter: cloudflare({
+		workerEntryPoint: {
+			path: 'src/worker.ts',
+			namedExports: ['MyDurableObject']
+		}
+	}),
+});
+```
+
+You can then create a custom entrypoint file at the specified path, which can include any additional imports or exports you need for your Workers setup. This example configures a custom entry file that registers a Durable Object and a queue handler:
+
+```ts title="src/worker.ts" "handle(manifest, app, request, env, ctx)"
+import type { SSRManifest } from 'astro';
+
+import { App } from 'astro/app';
+import { handle } from '@astrojs/cloudflare/handler'
+import { DurableObject } from 'cloudflare:workers';
+
+class MyDurableObject extends DurableObject<Env> {
+	constructor(ctx: DurableObjectState, env: Env) {
+		super(ctx, env)
+	}
+}
+
```

Comment:
Do we actually want to use a more practical example here, or do we trust users that want to use DurableObjects know how they work, since it is an advanced use-case anyways?

Reply:
I just used the example from your changelog, which I think is fine. It's just showing how it's done.